blockchain/interface.go

```diff
@@ -58,3 +58,8 @@ type FinalityProvider interface {
 	CheckShardFinalization(ctx context.Context, targetShardID uint32, maxNoncesDelta uint64) error
 	IsInterfaceNil() bool
 }
+
+type AddressGenerator interface {
```

Comment:
missing comment

Reply:
added

Reply:
able to generate smart contract addresses?